Documentation/4.7.0 release notes (#63)
FirelyTeam/firely-docs-firely-server: releasenotes/releasenotes.rst

.. _vonk_releasenotes_470:

Release 4.7.0, Feb 1st, 20222
-----------------------------

.. attention::    
    With version 4.7.0, Firely Server migrated to .NET 6.0. In order to run the binaries, `ASP.NET Core Runtime 6.x <https://dotnet.microsoft.com/en-us/download/dotnet/6.0>`_ needs to be installed.

Feature
^^^^^^^

#. BulkDataExport is now supported for MongoDB as well. Get started with the :ref:`Bulk Data Export documentation<feature_bulkdataexport>`.
#. Circular references in transaction bundles are now supported. Bundles of type ``transaction`` and ``batch`` are permitted to contain resources referencing another resource within the same bundle. This also means that you can now cross reference ``PUT`` and ``POST`` entries.
#. An option to configure additional token issuers is now available. This is used in settings where the token issuer deviates from the token audience. This new setting replaces the existing ``AdditionalEndpointBaseAddresses``. The setting needs to be adjusted manually as it will not be migrated automatically. Please check the :ref:`configuration documentation <feature_accesscontrol_config>` on how to use it.

> #. Firely Server now supports receiving document bundles on the transaction base endpoint. Firely Server will extract the narrative of document bundles and store this within a DocumentReference resource. Read more about it in the :ref:`documentation<restful_capabilities>`.

Reviewer: ```suggestion
#. Firely Server now supports receiving document bundles on the base endpoint. Firely Server will extract the narrative of document bundles and store this within a DocumentReference resource. Read more about it in the :ref:`documentation<restful_documenthandling>`.
```
I'd leave the word 'transaction' out here. As it has nothing to do with FHIR transactions, it may cause confusion. I also updated the reference, as 'restful_capabilities' is about the /metadata request.

The default FHIR version of the `public Firely Server endpoint <https://server.fire.ly/>`_ is now R4.

.. _vonk_releasenotes_470:

Release 4.7.0, Feb 1st, 20222
-----------------------------

.. attention::    
    With version 4.7.0, Firely Server migrated to .NET 6.0. In order to run the binaries, `ASP.NET Core Runtime 6.x <https://dotnet.microsoft.com/en-us/download/dotnet/6.0>`_ needs to be installed.

Feature
^^^^^^^

> #. BulkDataExport is now supported for MongoDB as well. Get started with the :ref:`Bulk Data Export documentation<feature_bulkdataexport>`.

Reviewer: Did you check all the existing BDE docs to see if something is said about MongoDB support? E.g. the [BDE page](https://docs.fire.ly/projects/Firely-Server/en/latest/features/bulkdataexport.html) has a note saying that we don't support it.

Reviewer: Also check all related settings documentation.

Author: Will address this in a separate PR, see VONK-3137

> #. Circular references in transaction bundles are now supported. Bundles of type ``transaction`` and ``batch`` are permitted to contain resources referencing another resource within the same bundle. This also means that you can now cross reference ``PUT`` and ``POST`` entries.

Reviewer: This also affects https://docs.fire.ly/projects/Firely-Server/en/latest/features/restful_api.html#transaction

Author: Thanks, I will address this in a separate PR, see VONK-3137 👍

> #. An option to configure additional token issuers is now available. This is used in settings where the token issuer deviates from the token audience. This new setting replaces the existing ``AdditionalBaseEndpoints``. The setting needs to be adjusted manually as it will not be migrated automatically.

Reviewer: Then also adjust https://docs.fire.ly/projects/Firely-Server/en/latest/security/accesscontrol.html#configuration
BTW: the existing setting is called `AdditionalEndpointBaseAddresses` according to that page.

Author: Thanks, I will address this in a separate PR, see VONK-3137 👍

> #. Firely Server now supports receiving document bundles on the transaction base endpoint. Firely Server will extract the narrative of document bundles and store this within a DocumentReference resource. 

Reviewer: That deserves a page in the Features section. @alexzautke : would that be something for you to add?

Author: Done!

> #. Added support for transforming SMART scopes issued by Azure Active Directory.

Reviewer: Merge PR #62 into this and refer to its new contents.

Author: Done!

#. Firely Server will now recognize the ``name`` claim in JSON Web Tokens and also include its content in the logs.

> #. It is now possible to provide the Firely Server license via an environment variable.

Reviewer: Merge PR #61 into this and refer to its new contents

Author: Done, thank you 👍

Plugins
^^^^^^^

.. attention::    

>     Please ensure that your plugins for Firely Server are targeting at least ``net6.0``. Plugins that are targeting ``netstandard2.0`` or ``netstandard2.1`` do not need to be updated.

Reviewer 2: This is somewhat confusing to me. Shouldn't it state that plugins that target netstandard 2.0/2.1 or netcore3.1 do not need to be updated and that new plugins can now also (optionally) target net6.0?

Reviewer: Agree.

Author: Actually, we can remove the entire warning. Nothing changes for plugin creators as they are targeting netstandard xx anyway, which is compatible with .NET 2.0 and above: https://docs.microsoft.com/en-us/dotnet/standard/net-standard?tabs=net-standard-2-0

> #. BulkDataExport interfaces were made publicly available in order to provide these to Firely Server's facade implementers.

Reviewer: @frank-olthuijsen : when your documentation is ready, make a link from here to it.

Reviewer 2: ```suggestion
The Bulk Data Export page now has a section on :ref:BDE for facades<feature_bulkdataexport_facade>
```

> #. Upgraded the .NET SDK to 3.7.0. Please review its release notes for changes.

Reviewer: Create a direct link to those release notes for convenience to the reader.

Author: Done, thank you 👍

Logging improvements
^^^^^^^^^^^^^^^^^^^^

> #. Error messages including information about authorization validation and authentication requests are now enriched with user information if ``ShowAuthorizationPII`` is enabled.

Reviewer: Where do you set that? Is that included in the settings on the relevant pages?

Author: Slightly outside the scope of this ticket but I added it to VONK-3137 and will later link to it 👍

> #. Authorization/Authentication logging messages are now enriched with more information when logging is set to ``debug``.

Reviewer: ```suggestion
#. Authorization/Authentication logging messages are now enriched with more information when logging level is set to ``Debug``.
```

Reviewer: And for which namespace should you set it to `Debug`?

Author: Added `Vonk.Smart` 🙂

> #. In case SSL is activated, but a ``.pfx`` file could not be found, Firely Server will now log this error more explicitly. 

Reviewer: ```suggestion
#. In case :ref:`SSL is activated<configure_hosting>`, but the ``.pfx`` file configured in ``CertificateFile`` could not be found, Firely Server will now log this error more explicitly. 
```
